functorch/functorch/experimental/cond.py

```diff
@@ -117,21 +123,17 @@ def cond_autograd(pred, true_fn, false_fn, *operands):
     return cond(pred, true_fn, false_fn, *operands)
 
 
-def python_fallback(op):
-    def inner(pred, true_fn, false_fn, operands):
-        mode = torch._C._get_torch_dispatch_mode()
-        assert (mode is not None), "Mode should always be enabled for python fallback key"
-        with suspend_mode(mode):
-            res = trace_cond(mode, op, pred, true_fn, false_fn, operands)
-        return res
-
-    return inner
+@cond.py_impl(DispatchKey.Python)
+@cond.py_impl(ProxyTorchDispatchMode)
```

Comment:
@ezyang thoughts on explicit mode registration? the idea is that if we see dispatch for an op, we assert that the given op is registered for the given mode. 

Should this be exclusion? Also we do not dispatch on modes yet, only keys, so we must assert that the impls match. Should we ever relax this assertion in the future?

Reply:
Actually, I have a better idea here...